drivers/gles3/shaders/canvas.glsl

```diff
@@ -278,6 +278,18 @@ float map_ninepatch_axis(float pixel, float draw_size, float tex_pixel_size, flo
 			ratio = mod(ratio * scale, 1.0);
 			// Scale to source texture.
 			return (margin_begin + ratio * dst_area) * tex_pixel_size;
+		} else if (np_repeat == 3) { // Center.
+			float src_area = draw_size - margin_begin - margin_end;
+			float dst_area = tex_size - margin_begin - margin_end;
+			float center = (src_area - dst_area) / 2.0;
+			if (pixel <= margin_begin + center) {
+				return (margin_begin + 1) * tex_pixel_size; // Stretch begin.
+			}
+			if (pixel >= draw_size - margin_end - center) {
+				return (tex_size - margin_end - 1) * tex_pixel_size; // Stretch end.
+			}
+
+			return (pixel - center) * tex_pixel_size;
```

Comment:
After removing the `+ 1` / `- 1` parts so the stretching would be continous this should be equivalent I think:
```suggestion
			return clamp(pixel - (draw_size - tex_size) / 2.0, margin_begin, tex_size - margin_end) * tex_pixel_size;
```

Reply:
I am stumped this received attention after such a long while. I appreciate showcasing the issue and the solution right away, thanks.